lsp-fsharp.el

```diff
@@ -179,14 +171,46 @@ disable if `--backgorund-service-enabled' is not used"
                     ".exe")))
     (expand-file-name (concat "fsautocomplete" file-ext) lsp-fsharp-server-install-dir)))
 
+(defun lsp-fsharp--version-list-latest (lst)
+  "Return latest version from LST (if any)."
+  (->> lst
+       (-map (lambda (x) (car (s-split " " x))))
+       (-filter (lambda (x) (> (length x) 0)))
+       (-sort (lambda (a b) (not (version<= (substring a 1)
+                                            (substring b 1)))))
+       cl-first))
+
+(defun lsp-fsharp--fetch-json (url)
+  "Retrieve and parse JSON from URL."
+  (with-temp-buffer
+    (url-insert-file-contents url)
+    (let ((json-false :false))
+      (json-read))))
+
+(defun lsp-fsharp--latest-version-from-github ()
+  "Return latest version of the server available from github."
+   (lsp-fsharp--version-list-latest
+    (seq-map (lambda (elt) (s-trim (cdr (assq 'name elt))))
+             (lsp-fsharp--fetch-json "https://api.github.com/repos/fsharp/FsAutoComplete/releases"))))
+
+(defun lsp-fsharp--server-download-url (version)
+  "Return url for .zip file to download for given VERSION, depending on lsp-fsharp-server-runtime."
```

Comment:
`https://github.com/%s/%s/releases/latest/download/%s`
If you just want latest version, this url is enough.
It save one web query which can be quite slow.